### PR TITLE
layers: Add sync2 support to simple validators

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
Update debug_printf, gpu_validation and best_practices to support
VK_KHR_synchronization2 entry points.

Change-Id: Icb742d14fbadbc47e2b7d8bb683c03dbd4ee97c5

### CODE REVIEW

File: layers/debug_printf.cpp [hunk 1]
```diff
@@ -637,6 +656,41 @@ void DebugPrintf::PostCallRecordQueueSubmit(VkQueue queue, uint32_t submitCount,
...
+        const VkSubmitInfo2KHR *submit = &pSubmits[submit_idx];
+        for (uint32_t i = 0; i < submit->commandBufferInfoCount; i++) {
+            auto cb_node = GetCBState(submit->pCommandBufferInfos[i].commandBuffer);
+            if (GetBufferInfo(cb_node->commandBuffer).size()) buffers_present = true;
+            for (auto secondaryCmdBuffer : cb_node->linkedCommandBuffers) {
+                if (GetBufferInfo(secondaryCmdBuffer->commandBuffer).size()) buffers_present = true;
+            }
+        }
+    }
+    if (!buffers_present) return;
```
Comment (Tony-LunarG) [inline layers/debug_printf.cpp:676]: Any chance PostCallRecordQueueSubmit and PostCallRecordRecordQueueSubmit2 could share common code, perhaps in a shared utility?

Comment (jeremyg-lunarg) [reply layers/debug_printf.cpp:676]: probably, I'll give it a try in a sec.

**Review by ncesario-lunarg — APPROVED** [review]
This LGTM. If it's possible to consolidate code as Tony suggested that would be great, but I'm not familiar enough with those functions to know if that makes sense or not in this case.

**Comment by jeremyg-lunarg** [comment]
> This LGTM. If it's possible to consolidate code as Tony suggested that would be great, but I'm not familiar enough with those functions to know if that makes sense or not in this case.

I consolidated down a bit with the CommandBufferNeedsProcessing() and ProcessCommandBuffer() methods. Going further than that gets tricky because of the differences in how the VkSubmitInfo and VkSubmitInfo2KHR structs are laid out.

**Pull request merged**